tests/pytests/test_aux_save2.py

```diff
@@ -0,0 +1,278 @@
+    env.stop()
+    # Restart with modules
+    env.start()
+    # doc1 should exist
+    env.expect('HGET', 'doc1', 't').equal('lion')
+    # dict1 should exist
+    res = env.cmd('FT.DICTDUMP', 'dict1')
+    env.assertEqual(res, ['bar', 'baz', 'hakuna matata'])
+    # dict2 does not exist, but FT.DICTDUMP returns an empty list
+    env.expect('FT.DICTDUMP', 'dict2').equal([])
+
+@skip(cluster=True)
+def testLoadRdbWithEmptySpellcheckDict(env):
+    # Test loading an RDB with 3 dictionaries:
+    # empty_dict1 and empty_dict2 are empty dictionaries
+    # dict is a non-empty dictionary, containing two items: ['hello', 'hola']
+
+    env = Env()
+    skipOnExistingEnv(env)
+    dbFileName = env.cmd('config', 'get', 'dbfilename')[1]
+    dbDir = env.cmd('config', 'get', 'dir')[1]
+    rdbFilePath = os.path.join(dbDir, dbFileName)
+    if not downloadFiles(RDBS):
+        if CI:
+            env.assertTrue(False)  ## we could not download rdbs and we are running on CI, let fail the test
+        else:
+            env.skip()
+            return
+
+    for fileName in RDBS:
+        env.stop()
+        filePath = os.path.join(REDISEARCH_CACHE_DIR, fileName)
+        try:
+            os.unlink(rdbFilePath)
+        except OSError:
+            pass
+        os.symlink(filePath, rdbFilePath)
+        env.start()
+
+        # Check that the non-empty dictionary is loaded
+        res = env.cmd('FT.DICTDUMP', 'dict')
+        env.assertEqual(res, ['hello', 'hola'])
+
+        # File size after saving the RDB is smaller than the original file size
+        # because the empty dictionaries are not saved
+        filesize = os.path.getsize(rdbFilePath)
+        env.assertEqual(filesize, 205)
+        env.cmd('SAVE')
+        filesize_after_save = os.path.getsize(rdbFilePath)
+        env.assertGreater(filesize, filesize_after_save)
+
+@skip(cluster=True, no_json=True, asan=True)
+def testLoadRdbWithoutSuggestionData(env: Env):
+    env.expect('FT.SUGADD', 'sug', 'hakuna matata', '1').equal(1)
+    env.expect('FT.SUGADD', 'sug', 'hakuna', '1').equal(2)
+    env.expect('HSET', 'doc1', 't', 'lion').equal(1)
+    res = env.cmd('FT.SUGGET', 'sug', 'hakuna')
+    env.assertEqual(res, ['hakuna', 'hakuna matata'])
+    env.expect('FT.SUGDEL', 'sug', 'hakuna matata').equal(1)
+    env.expect('FT.SUGDEL', 'sug', 'hakuna').equal(1)
+    env.expect('FT.SUGLEN', 'sug').equal(0)
+    # sug should exist, the key is not deleted when the last item is removed
+    env.expect('EXISTS', 'sug').equal(1)
+    # Save state to RDB
+    env.stop()
+    # Restart without modules
+    env.assertEqual(len(env.envRunner.modulePath), 2)
+    env.assertEqual(len(env.envRunner.moduleArgs), 2)
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.modulePath.pop()
+    env.envRunner.moduleArgs.pop()
+    env.envRunner.masterCmdArgs = env.envRunner.createCmdArgs('master')
+    # Attempt to load RDB fails because the RDB contains an empty
```

Comment:
If we change the suggestion dictionary to be deleted when its last entry is removed - this rdb could be reloaded